File: java/day01/Day01.java
void main(String[] args) {
    File file = new File("java/day01/day01.txt");
    try (FileReader fileReader = new FileReader(file)) {
        int zeros = 0;
        int current = 50;
        for (String line : fileReader.readAllLines()) {
            int turns = Integer.parseInt(line.substring(1)) % 100;
            if (line.startsWith("L")) {
                current -= turns;
                if (current < 0) {
                    current = current + 100;
                }
            } else {
                current += turns;
                if (current >= 100) {
                    current = current - 100;
                }
            }

            if (current == 0) {
                zeros++;
            }
            System.out.printf("%s --> %d%n", line, current);
        }
        System.out.println("Zeros: " + zeros);
    } catch (IOException e) {
        throw new RuntimeException(e);
    }
}

File: python/day01/day01.py
def solve() -> tuple[int, int]:
    current = 50
    zeros, passing_over_zero = 0, 0

    with open('../../input/day01/day01.txt') as f:
        for line in f.readlines():
            direction = -1 if line[0] == "L" else 1
            turns = int(line[1:])

            if direction < 0:
                to_next_zero = current or 100
            else:
                to_next_zero = 100 - current

            current = (current + direction * turns) % 100

            if current == 0:
                zeros += 1
            if turns >= to_next_zero:
                passing_over_zero += (turns - to_next_zero) // 100 + 1

    return zeros, passing_over_zero


if __name__ == '__main__':
    print(solve())
